File: app/src/main/java/com/frankly/randomdog/randomdogapp/RandomDogApplication.java
package com.frankly.randomdog.randomdogapp;

import android.app.Application;
import android.content.Context;

import com.frankly.randomdog.bitmaputil.BitmapCache;
import com.frankly.randomdog.sharedpreferences.SharedPref;


public class RandomDogApplication extends Application {
    static BitmapCache  cache = null;
    static SharedPref pref = null;
    static Context context = null;

    @Override
    public void onCreate() {
        super.onCreate();
        cache = new BitmapCache(BitmapCache.getCacheSize());
        context = getApplicationContext();
        pref = new SharedPref(context);

    }

    public static BitmapCache getCache() {
        return cache;
    }

    public static SharedPref getPref() {
        return pref;
    }

    public static Context getContext() {
        return context;
    }
}

File: app/src/main/java/com/frankly/randomdog/randomdogapp/RandomDogApplication.kt
package com.frankly.randomdog.randomdogapp

import android.app.Application
import android.content.Context
import com.frankly.randomdog.bitmaputil.BitmapCache
import com.frankly.randomdog.sharedpreferences.SharedPref

/**
 * @author: ArshVardhanAtreya <arshvardhan></arshvardhan>@yahoo.co.in>
 */
class RandomDogApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        lruCache = BitmapCache(BitmapCache.getCacheSize())
        appContext = applicationContext
        sharedPref = SharedPref(appContext)
    }

    companion object {
        var lruCache: BitmapCache? = getCache()
        var sharedPref: SharedPref? = getPref()
        var appContext: Context? = getContext()

        fun getCache(): BitmapCache? {
            return lruCache
        }

        fun getPref(): SharedPref? {
            return sharedPref
        }

        fun getContext(): Context? {
            return appContext
        }
    }
}